definitions/features/container_gateway_database.rb

```diff
@@ -0,0 +1,40 @@
+class Features::ContainerGatewayDatabase < ForemanMaintain::Feature
+  CONTAINER_GATEWAY_DB_CONFIG = '/etc/foreman-proxy/settings.d/container_gateway.yml'.freeze
+
+  include ForemanMaintain::Concerns::BaseDatabase
+  include ForemanMaintain::Concerns::DirectoryMarker
+
+  metadata do
+    label :container_gateway_database
+
+    confine do
+      file_nonzero?(CONTAINER_GATEWAY_DB_CONFIG)
+    end
+  end
+
+  def configuration
+    @configuration || load_configuration
+  end
+
+  def services
+    [
+      system_service('postgresql', 10, :component => 'container_gateway',
+        :db_feature => feature(:container_gateway_database)),
+    ]
+  end
+
+  private
+
+  def load_configuration
+    config = YAML.load(File.read(CONTAINER_GATEWAY_DB_CONFIG))
+    @configuration = {}
+    connection_string = config[:db_connection_string]
+    if connection_string
+      uri = URI.parse(connection_string)
+      @configuration['connection_string'] = connection_string
+      @configuration['user'] = 'foreman-proxy'
+      @configuration['database'] = uri.path[1..]
+    end
```

Comment:
So looking at https://github.com/theforeman/puppet-foreman_proxy/pull/835, the `connection_string` is something like `postgres:///container_gateway` (for a local setup) or `postgres://configurableuser:changeme@test.example.com:5432/container_gateway` (for a remote DB, supported or not).

Now, the other code in this PR depends on the `user` definition whether it switches to that (shell) user or not.
That switching is unnecessary for the remote-DB case!

```suggestion
    connection_string = config[:db_connection_string]
    if connection_string
      uri = URI.parse(connection_string)
      @configuration['connection_string'] = connection_string
      @configuration['user'] = 'foreman-proxy' unless uri.host
      @configuration['database'] = uri.path[1..]
    end
```